src/test/archive/patch_archive_test/patch_archive_test.ml

```diff
@@ -80,33 +62,21 @@ let main ~db_uri ~network_data_folder () =
         (* never remove last block as missing-block-guardian can have issues when patching it
            as it patching only gaps
         *)
-        Random.int (List.length extensional_files - 1) )
+        Random.int (List.length extensional_files - 2) + 1 )
```

Comment:
What about first block? Is it true that we always have the first block precomputed and read from input?

Reply:
This is going to be error if we only have 2 files. I understand that's unlikely but could we somehow deal with it? A failwith would be fine, or in an `Or_error` monad would be better.